From: Jorge Lopez <jorgealtxwork@gmail.com>
Date: Mon, 21 Aug 2023 09:42:05 -0500
Subject: platform/x86: hp-bioscfg: Update steps order list elements are
 evaluated
References: jsc#PED-13019
Git-commit: 559eed7776c6557acf58e86c48ddaed9301550fd
Patch-mainline: 6.6-rc1

Update steps how order list elements data and elements size are
evaluated.

Signed-off-by: Jorge Lopez <jorge.lopez2@hp.com>
Link: https://lore.kernel.org/r/20230821144205.13529-1-jorge.lopez2@hp.com
Reviewed-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/platform/x86/hp/hp-bioscfg/order-list-attributes.c |   10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

--- a/drivers/platform/x86/hp/hp-bioscfg/order-list-attributes.c
+++ b/drivers/platform/x86/hp/hp-bioscfg/order-list-attributes.c
@@ -258,7 +258,6 @@ static int hp_populate_ordered_list_elem
 				eloc++;
 			break;
 		case ORD_LIST_ELEMENTS:
-			size = ordered_list_data->elements_size;
 
 			/*
 			 * Ordered list data is stored in hex and comma separated format
@@ -270,17 +269,14 @@ static int hp_populate_ordered_list_elem
 
 			part_tmp = tmpstr;
 			part = strsep(&part_tmp, COMMA_SEP);
-			if (!part)
-				strscpy(ordered_list_data->elements[0],
-					tmpstr,
-					sizeof(ordered_list_data->elements[0]));
 
-			for (olist_elem = 1; olist_elem < MAX_ELEMENTS_SIZE && part; olist_elem++) {
+			for (olist_elem = 0; olist_elem < MAX_ELEMENTS_SIZE && part; olist_elem++) {
 				strscpy(ordered_list_data->elements[olist_elem],
 					part,
 					sizeof(ordered_list_data->elements[olist_elem]));
-				part = strsep(&part_tmp, SEMICOLON_SEP);
+				part = strsep(&part_tmp, COMMA_SEP);
 			}
+			ordered_list_data->elements_size = olist_elem;
 
 			kfree(str_value);
 			str_value = NULL;
